File: 1422-maximum-score-after-splitting-a-string/1422-maximum-score-after-splitting-a-string.cpp
class Solution {
public:
    int maxScore(string s) {
        int n = s.length();
        int leftZeros = 0, rightOnes = 0;
        int maxScore = 0; 
        for (char c : s) {
            if (c == '1') {
                rightOnes++;
            }
        }
        for (int i = 0; i < n - 1; i++) {
            if (s[i] == '0') {
                leftZeros++;
            }
            if (s[i] == '1') {
                rightOnes--;
            }
            maxScore = max(maxScore, leftZeros + rightOnes);
        }

        return maxScore;
    }
};

File: 1422-maximum-score-after-splitting-a-string/1422-maximum-score-after-splitting-a-string.java
class Solution {
    public int maxScore(String s) {
        int leftZeros = 0, rightOnes = 0, maxScore = 0;
        int n = s.length();
        for (char c : s.toCharArray()) {
            if (c == '1')
                rightOnes++;
        }
        for(int i=0; i<n-1;i++){
            if(s.charAt(i) == '0')
                leftZeros++;
            if(s.charAt(i)=='1')
                rightOnes--;
            maxScore = Math.max(maxScore, leftZeros + rightOnes);
        }
        return maxScore;
    }
    
}